keps/sig-node/1287-in-place-update-pod-resources/README.md

```diff
@@ -659,6 +696,44 @@ Pod Status in response to user changing the desired resources in Pod Spec.
   in ContainerStatus.Resources to update ContainerStatuses[i].Resources.Limits
   for that Container in the Pod's Status.
 
+#### Kubelet Restart Analysis
+
+Analysis of Kubelet restarts happening at various points of resize, and how recovery happens.
+Impacts of a restart outside of resource configuration are out of scope.
+
+1. Kubelet Admits a new pod
+  - Resource allocation checkpointed before sending the pod to the pod workers
+  - Restart before checkpointing: pod goes through admission again as if new
+  - Restart after checkpointing: pod goes through admission using the allocated resources
+1. Kubelet creates a container
+  - Resources acknowledged after CreateContainer call succeeds
+  - Restart before acknowledgement: Kubelet issues a superfluous UpdatePodResources request
```

Comment:
Just a note: cc/ @samuelkarp and @mikebrow to ensure UpdatePodResources call is idempotent since I roughly remembered there was an bug at CRI layer on this lately.

Reply:
I'm going to continue to argue that kubelet should be periodically re-asserting the resources it wants, so this absolutely needs to be idempotent :)

Also, I think we should just re-assert at startup for all pods and containers.

Reply:
+1 with you above. We should address this, but not a block for this KEP.

Reply:
Idempotence is mentioned under the CRI changes (and actuating resources). Resync is under future enhancements.